src/image.js

```diff
@@ -0,0 +1,121 @@
+require("babel-polyfill");
+
+import gm from "gm";
+import log from "./log";
+import uuid from "uuid";
+
+const im = gm.subClass({imageMagick: true});
+
+// Wrap these calls in promises so we can use async/await
+const promiseWrite = (client, file) => {
+  return new Promise((resolve, reject) => client.write(file, err => err ? reject(err) : resolve()));
+};
+const promiseSize = (client) => {
+  return new Promise((resolve, reject) => client.size((err, data) => err ? reject(err) : resolve(data)));
+};
+
+// Interlacing for png isn't efficient (both in filesize as render performance), so we only do it for jpg
+const interlace = async(client, params) => {
+  if (params.mime === 'image/jpeg') {
+    return client.interlace('Line');
+  }
+  return client;
+};
+
+// When cropping, we size the image first to fix completely within the bounding box
+// Then we crop the requested size, and center to the center of the image
+const crop = async(client, params) => {
+  // Resize the image to fit within the bounding box
+  // The ^ ensures the images is resized while maintaining the ratio
+  // However the sizes are treated as minimum values instead of maximum values
+  client = client.resize(params.width, params.height, '^');
+  const tmpFile = `/tmp/${uuid.v4()}`;
+  try {
+    await promiseWrite(client, tmpFile);
```

Comment:
I don't fully get why you need the temp file. Can you elaborate? 

You also don't seem to clean up the file. Not all systems are configured to regularly clean up the tmp folder (I think in fact most aren't, only on startup), so you might want to do this manually.